vignettes_src/orderly.Rmd

```diff
@@ -0,0 +1,104 @@
+---
+title: "Using didehpc to run orderly tasks"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+    toc_depth: 2
+vignette: >
+  %\VignetteIndexEntry{Using didehpc to run orderly tasks}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+<!-- HEADER -->
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  error = FALSE,
+  comment = "#>"
+)
+orderly_root <- orderly:::prepare_orderly_example("demo")
+```
+
+When using [orderly](https://github.com/vimc/orderly) some tasks might be slow to run and would be better suited to running on a remote machine. You can use the cluster and [orderly bundles](https://www.vaccineimpact.org/orderly/articles/bundles.html) to achieve this. If your orderly instance has an [OrderlyWeb](https://github.com/vimc/orderly-web) remote then you should favour running tasks on the configured remote via `orderly::orderly_run_remote` (unless it is too slow even for the OrderlyWeb remote). If your orderly instance is using a [sharepoint remote](https://github.com/vimc/orderly.sharepoint) or has no remote then you can use the cluster to run your slow running tasks.
+
+## Running a task on the cluster
+
+To run a single task or report on the cluster start by initialising your bundle on a network drive. For example on Linux:
+
+```{r}
+root <- "~/net/home/contexts"
+path_bundles <- file.path(root, "bundles")
+
+## Create bundle
+bundle <- orderly::orderly_bundle_pack(path_bundles, 
+                                       "minimal", 
+                                       root = orderly_root)
+```
+
+Then setup the cluster with required packages, you will need `orderly` at a minimum. You can use the list of packages from the `orderly.yml` e.g. for report `internal-2017-population-TUV-MHL`
+
+```{r}
+orderly_packages <- yaml::read_yaml(
+  file.path(orderly_root, "src/minimal/orderly.yml"))$packages
+packages <- list(loaded = c("orderly", orderly_packages))
+config <- didehpc::didehpc_config(workdir = root)
+ctx <- context::context_save(root, packages = packages)
+obj <- didehpc::queue_didehpc(ctx, config = config)
+```
+
+The orderly task can then be run via `orderly::orderly_bundle_run`, being careful to make sure the paths are relative to the `workdir` passed in `didehpc_config`
```

Comment:
Hmm, this seems complicated. It might be worth being a bit more explicit somewhere about the setup:

* network path `N` that we'll use for running the bundles
* working directory for the session which will be `N` or some parent directory
* (possibly) non-network path that we'll be treating as our canonical repository

Reply:
Even though I wrote this I don't have a great idea of what is going on here! We have to provide a `workdir` arg because this would otherwise run in a temporary directory which would be deleted once the cluster process stops. But is that argument relative to the bundle directory or the working directory (I'd guess the latter?). Should we use "input" then rather than "bundles" so that we have:

* `<workdir>/input/<id>.zip`
* `<workdir>/output/<id>.zip`

People will typically follow exactly the docs, and this might be a nice pattern